homeassistant/components/solarlog/sensor.py

```diff
@@ -21,200 +22,219 @@
 
 
-@dataclass(frozen=True)
-class SolarLogSensorEntityDescription(SensorEntityDescription):
-    """Describes Solarlog sensor entity."""
+@dataclass(frozen=True, kw_only=True)
+class SolarLogCoordinatorSensorEntityDescription(SensorEntityDescription):
+    """Describes Solarlog coordinator sensor entity."""
+
+    value_fn: Callable[[SolarlogData], StateType | datetime | None]
+
 
-    value_fn: Callable[[float | int], float] | Callable[[datetime], datetime] = (
-        lambda value: value
-    )
+@dataclass(frozen=True, kw_only=True)
+class SolarLogInverterSensorEntityDescription(SensorEntityDescription):
+    """Describes Solarlog inverter sensor entity."""
 
+    value_fn: Callable[[InverterData], float | None]
 
-SOLARLOG_SENSOR_TYPES: tuple[SolarLogSensorEntityDescription, ...] = (
-    SolarLogSensorEntityDescription(
+
+SOLARLOG_SENSOR_TYPES: tuple[SolarLogCoordinatorSensorEntityDescription, ...] = (
+    SolarLogCoordinatorSensorEntityDescription(
         key="last_updated",
         translation_key="last_update",
         device_class=SensorDeviceClass.TIMESTAMP,
+        value_fn=lambda data: data.last_updated,
     ),
-    SolarLogSensorEntityDescription(
+    SolarLogCoordinatorSensorEntityDescription(
         key="power_ac",
         translation_key="power_ac",
         native_unit_of_measurement=UnitOfPower.WATT,
         device_class=SensorDeviceClass.POWER,
         state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.power_ac,
     ),
-    SolarLogSensorEntityDescription(
+    SolarLogCoordinatorSensorEntityDescription(
         key="power_dc",
         translation_key="power_dc",
         native_unit_of_measurement=UnitOfPower.WATT,
         device_class=SensorDeviceClass.POWER,
         state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.power_dc,
     ),
-    SolarLogSensorEntityDescription(
+    SolarLogCoordinatorSensorEntityDescription(
         key="voltage_ac",
         translation_key="voltage_ac",
         native_unit_of_measurement=UnitOfElectricPotential.VOLT,
         device_class=SensorDeviceClass.VOLTAGE,
         state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.voltage_ac,
     ),
-    SolarLogSensorEntityDescription(
+    SolarLogCoordinatorSensorEntityDescription(
         key="voltage_dc",
         translation_key="voltage_dc",
         native_unit_of_measurement=UnitOfElectricPotential.VOLT,
         device_class=SensorDeviceClass.VOLTAGE,
         state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.voltage_dc,
     ),
-    SolarLogSensorEntityDescription(
+    SolarLogCoordinatorSensorEntityDescription(
         key="yield_day",
         translation_key="yield_day",
         native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
         device_class=SensorDeviceClass.ENERGY,
-        value_fn=lambda value: round(value / 1000, 3),
+        value_fn=lambda data: round(data.yield_day / 1000, 3),
```

Comment:
Also for a followup, unless these entities are really noisy, remove the round and add the previous precision to `suggested_display_precision`